### ISSUE TITLE
Apple Pass Did not open successfully.

### ISSUE THREAD

**Issue by HammadUllahKhan12** [alexandercerutti/passkit-generator#82]
This is my pass.json

```
{
	"formatVersion": 1,
	"passTypeIdentifier": "pass.com.parksuiteapp.parksuite",
	"serialNumber": "1968447676928229592",
	"teamIdentifier": "6U93C32K9G",
	"barcodes": [
		{
			"message": "123456789",
			"format": "PKBarcodeFormatQR",
			"messageEncoding": "iso-8859-1"
		}
	],
	"barcode": {
		"message": "123456789",
		"format": "PKBarcodeFormatQR",
		"messageEncoding": "iso-8859-1"
	},
	"organizationName": "Github Inc.",
	"description": "A Booking pass",
	"foregroundColor": "rgb(255, 255, 255)",
	"backgroundColor": "rgb(253, 123, 35)",
	"coupon": {}
}
```

**Comment by HammadUllahKhan12** [comment]
For the certificates. 

> This is my pass.json
> 
> ```
> {
> 	"formatVersion": 1,
> 	"passTypeIdentifier": "pass.com.parksuiteapp.parksuite",
> 	"serialNumber": "1968447676928229592",
> 	"teamIdentifier": "6U93C32K9G",
> 	"barcodes": [
> 		{
> 			"message": "123456789",
> 			"format": "PKBarcodeFormatQR",
> 			"messageEncoding": "iso-8859-1"
> 		}
> 	],
> 	"barcode": {
> 		"message": "123456789",
> 		"format": "PKBarcodeFormatQR",
> 		"messageEncoding": "iso-8859-1"
> 	},
> 	"organizationName": "Github Inc.",
> 	"description": "A Booking pass",
> 	"foregroundColor": "rgb(255, 255, 255)",
> 	"backgroundColor": "rgb(253, 123, 35)",
> 	"coupon": {}
> }
> ```
> 
> generate from this code .
> 
> ![image](https://user-images.githubusercontent.com/48718737/135426007-234104d6-8d36-4c69-a7d0-aae9eb56ee4f.png)

**Comment by alexandercerutti** [comment]
Hi @HammadUllahKhan12, you should debug it and check what's wrong. It is not possible for me to understand the issue by just the attached `pass.json`. Opening issues might happen for several reasons.

I can help you only if you have direct questions.
I've wrote a debug post in this repo's wiki you can follow: https://github.com/alexandercerutti/passkit-generator/wiki/Self-help

Let me know!

**Comment by alexandercerutti** [comment]
Please, try to use only one initial message per issue, because otherwise you bomb me out of notifications. You are also pleased to specify on which platform you are trying to open the pass. Requirements vary from platform to platform.

**Comment by HammadUllahKhan12** [comment]
hey @alexandercerutti I used node & express platform. Pass is generated successfully but when i try open it. It gives passes.pkpass could not opend.

**Comment by alexandercerutti** [comment]
Hi @HammadUllahKhan12, what is Console.app reporting?

**Comment by HammadUllahKhan12** [comment]
it is simple node.js app. API endpoint. that generate the pass @alexandercerutti

**Comment by alexandercerutti** [comment]
@HammadUllahKhan12 You have to debug it. Did you follow the guide I linked you?

**Comment by HammadUllahKhan12** [comment]
yes! @alexandercerutti i follow all the guide line and used the pass json file same as the example exampleBooking.pass. Double verified the keys it generated the pass successfully.  Can we have a way to check that which apple keys is causing the issue.

**Comment by alexandercerutti** [comment]
@HammadUllahKhan12 Yeah, but what do `Console.app` says when you try to open your pass? If a pass cannot be open, Console.app reports the full error, as I said in self-help guide I linked you

**Comment by HammadUllahKhan12** [comment]
@alexandercerutti i debug the issue on MAC. It gives these two in the logs.

`ERROR: Error Domain=PKPassKitErrorDomain Code=1 "The pass cannot be read because it isn’t valid." UserInfo={NSLocalizedDescription=The pass cannot be read because it isn’t valid., NSUnderlyingError=0x600000b3a580 {Error Domain=PKPassKitErrorDomain Code=1 "(null)"}}`

`PKZipArchiver: archive_read_next_header failed - Extra data overflow: Need 48575 bytes but only found 8377 bytes.
`

**Comment by alexandercerutti** [comment]
@HammadUllahKhan12 uh, that's really weird... Seems like it is being saved, maybe, with the wrong content-length, but I'm not sure. This never happened to me.

Are you trying to serve the PKPass saved on disk (the first image you sent to me)?
What happens if you try to serve the pass as-is, by piping the stream to the response, instead of saving it? (Look at examples folder for an example of what I'm saying).

Please note also that saving a pass to disk is a bad practice and it is discouraged by Apple itself.

You can also try another thing: rename the .pkpass to .zip and try to unzip it. Check which files are included inside the bundle and try to understand which might be the issue, also by making few generation attempts by excluding another file every time and see if the error changes.

Let me know

**Comment by alexandercerutti** [comment]
Hey there @HammadUllahKhan12, any news?

**Comment by HammadUllahKhan12** [comment]
@alexandercerutti I saved the pass in one endpoint.
  ```
 // Adding some settings to be written inside pass.json
    // Generate the stream .pkpass file stream
    const stream = await examplePass.generate()
    const data = await streamToBuffer(stream)
    // eslint-disable-next-line max-len
    // eslint-disable-next-line consistent-return
    await fs.writeFile(`${__dirname}/passes.pkpass`, data.toString(), err => {
      if (err) {
        return ApolloError('Error is in pass creation.')
      }
    })
    
```
& return it another endpoint.

**Comment by HammadUllahKhan12** [comment]
hey @alexandercerutt I converted the file but when i open it in MAC it gives this error.
<img width="327" alt="Screenshot 2021-10-12 at 6 56 05 PM" src="https://user-images.githubusercontent.com/48718737/136970002-cbaefb2b-87f6-4228-8d7d-d0fe09858379.png">
i

**Comment by alexandercerutti** [comment]
I'm unable to open it either by renaming it to .zip. It cannot recognize the file format.
There must be something wrong with zip creation, but I don't know what.
Which files you tried to insert in this pass?

**Comment by HammadUllahKhan12** [comment]
//   const examplePass = await createPass({
  //     model: `${__dirname}/apple/exampleBooking.pass`,
  //     certificates: {
  //       wwdr: `${__dirname}/apple/WWDR.pem`,
  //       signerCert: `${__dirname}/apple/signerCert.pem`,
  //       signerKey: {
  //         keyFile: `${__dirname}/apple/signerKey.pem`,
  //         passphrase: '12345678',
  //       },
  //     },
  //   })
  
  Model is same as example one.

**Comment by rahul2104** [comment]
PKZipArchiver: archive_read_next_header failed - Extra data overflow: Need 28519 bytes but only found 0 bytes.

i am getting this error can any one help what is the issue in pkpass
